### PR TITLE
Convert :workflow-runtime to KMM

### PR DESCRIPTION
**Pull request by bnvinay92**
- Replaces SendChannel#offer with SendChannel#trySend.
- org.jetbrains.kotlinx:kotlinx-coroutines-test:1.5.1 only ships a jvm artifact so bumped only this test dependency in commonTest to 1.6.1 (which ships artifacts for all platforms).

### CODE REVIEW

File: workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/RenderWorkflowInTest.kt [hunk 1]
```diff
@@ -34,40 +36,31 @@ import kotlin.test.assertTrue
...
+  private val testScope = TestScope(UnconfinedTestDispatcher())
 
   @Test fun `initial rendering is calculated synchronously`() {
     val props = MutableStateFlow("foo")
     val workflow = Workflow.stateless<String, Nothing, String> { "props: $it" }
     // Don't allow the workflow runtime to actually start.
-    expectedSuccessScope.pauseDispatcher()
-    val renderings = renderWorkflowIn(workflow, expectedSuccessScope, props) {}
+
+    val renderings = renderWorkflowIn(workflow, TestScope(), props) {}
```
Comment (steve-the-edwards) [inline workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/RenderWorkflowInTest.kt:54]: Can't we use `pausedTestScope` here?

File: workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/RenderWorkflowInTest.kt [hunk 2]
```diff
@@ -188,6 +180,7 @@ class RenderWorkflowInTest {
     }
     sink.send("unchanging state")
     sink.send("unchanging state")
+    pausedTestScope.advanceUntilIdle()
```
Comment (steve-the-edwards) [inline workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/RenderWorkflowInTest.kt:183]: why is this here? I don't think `pausedTestScope` is used in this test is it?

File: workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt [hunk 3]
```diff
@@ -133,19 +129,21 @@ internal class WorkflowRunnerTest {
...
-    dispatcher.resumeDispatcher()
-
     // The order in which props update and workflow update are processed is deterministic, based
     // on the order they appear in the select block in nextOutput.
-    val firstOutput = scope.async { runner.nextOutput() }
-      .getCompleted()
+    val firstOutput = with(scope.async { runner.nextOutput() }) {
+      scope.advanceUntilIdle()
+      getCompleted()
+    }
```
Comment (steve-the-edwards) [inline workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt:137]: We could throw this in a utility test extension function.

**Review by steve-the-edwards — APPROVED** [review]
Thanks @bnvinay92  - this is so exciting!

LGTM. @RBusarow also to take a look as he is tracking along with these close.y

File: workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt (hunk 3 above)
Comment (RBusarow) [reply workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt:137]: +1

There's some subtlety to this, so giving it a function with a good name would probably make it easier to read.

**Review by RBusarow — APPROVED** [review]
I had the same nitpicks as @steve-the-edwards, but overall this is great!

File: workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt (hunk 3 above)
Comment (bnvinay92) [reply workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkflowRunnerTest.kt]: Tried this: 268daedd27bf086efe4f39847112db9a7a41810b
